src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.GeneratedCodeTokenWalker.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.CodeDom.Compiler;
+
+namespace Microsoft.CodeAnalysis.Diagnostics
+{
+    internal abstract partial class AnalyzerDriver
+    {
+        private sealed class GeneratedCodeTokenWalker : SyntaxWalker
+        {
+            public GeneratedCodeTokenWalker()
+                : base(SyntaxWalkerDepth.Token)
+            {
+            }
+
+            public bool HasGeneratedCodeIdentifier { get; private set; }
+
+            public override void Visit(SyntaxNode node)
+            {
+                if (HasGeneratedCodeIdentifier)
+                    return;
+
+                base.Visit(node);
+            }
+
+            protected override void VisitToken(SyntaxToken token)
+            {
+                HasGeneratedCodeIdentifier |= string.Equals(token.ValueText, "GeneratedCode", StringComparison.Ordinal)
+                    || string.Equals(token.ValueText, nameof(GeneratedCodeAttribute), StringComparison.Ordinal);
+            }
```

Comment:
1. do we need to do the same for VB?
2. OMG, why d we have this code?  don't we already have an existin function to decide if code is generated or not?

Reply:
So we have GeneratedCodeUtilities.IsGeneratedCode...

Reply:
This PR affects both C# and VB

Reply:
in that case, shoudln't the be OrdinalIgnoreCase?  (and we should likely test for that too).

Reply:
(since the code tests for symbol equality, and VB would find the named type symbols to be equal even with different casing).

Reply:
I prefer to keep the faster fast path since that's what was already implemented. If we didn't already ship support for analyzer recognition of `[GeneratedCode]` I would be [strongly opposed](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/1094#issuecomment-126495130) to adding it at all.

Reply:
Ah, that's fair.  The previous code behaved the same way, so i'm fine with htis as is.